signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6cb508

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
panic
```
I1026 19:42:45.016882       1 server.go:68] Listening on http://0.0.0.0:9402
I1026 19:42:45.018889       1 leaderelection.go:174] attempting to acquire leader lease...
I1026 19:42:45.050322       1 leaderelection.go:184] successfully acquired lease kube-system/cert-manager-controller
I1026 19:42:45.050413       1 factory.go:58] NS does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Certificate"}
I1026 19:42:45.050478       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Issuer"}
I1026 19:42:45.050500       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"ClusterIssuer"}
I1026 19:42:45.050522       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 19:42:45.050534       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"extensions", Version:"v1beta1", Kind:"Ingress"}
I1026 19:42:45.050598       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"ClusterIssuer"}
I1026 19:42:45.050615       1 factory.go:58] NS does not exist on lookup for kube-system - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 19:42:45.050649       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Issuer"}
I1026 19:42:45.050664       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
E1026 19:42:45.085869       1 reflector.go:199] github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87: Failed to list *v1alpha1.ClusterIssuer: the server could not find the requested resource (get clusterissuers.certmanager.k8s.io)
I1026 19:42:45.151219       1 logs.go:41] starting worker
I1026 19:42:45.151298       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 19:42:45.151460       1 logs.go:41] starting worker
E1026 19:42:45.151473       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:290
/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:647
/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:321
/go/src/github.com/jetstack-experimental/cert-manager/pkg/issuer/acme/setup.go:59
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/sync.go:28
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:179
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:60
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:141
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:148
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2197
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6cb508]

goroutine 155 [running]:
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x126
panic(0x15ffe60, 0x2205470)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme.(*Client).doReg(0x0, 0x21ba740, 0xc4203cff80, 0x0, 0x0, 0x17b3255, 0x3, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:647 +0x128
github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme.(*Client).GetReg(0x0, 0x21ba740, 0xc4203cff80, 0x0, 0x0, 0x1, 0x14, 0xc4203143c0)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:321 +0x7e
github.com/jetstack-experimental/cert-manager/pkg/issuer/acme.(*Acme).Setup(0xc42015d980, 0x21ba740, 0xc4203cff80, 0x21baa00, 0xc42015d980, 0x0, 0x0, 0xc420237dc0, 0xc420416a60)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/issuer/acme/setup.go:59 +0x390
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Sync(0xc42031c140, 0x21ba740, 0xc4203cff80, 0xc4202cf500, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/sync.go:28 +0xcc
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).processNextWorkItem(0xc42031c140, 0x21ba740, 0xc4203cff80, 0xc420416a60, 0x20, 0xc420679d48, 0xc4202daf10)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:179 +0x234
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).(github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.processNextWorkItem)-fm(0x21ba740, 0xc4203cff80, 0xc420416a60, 0x20, 0x2, 0x2)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:60 +0x52
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).worker.func1(0xc42031c140, 0xc42001a720, 0x15749e0, 0xc420258a40, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:141 +0x2ba
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).worker(0xc42031c140, 0xc42001a720)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:148 +0xe5
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Run.func1()
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109 +0x33
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4203f2260)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4203f2260, 0x3b9aca00, 0x0, 0x1, 0xc42001a720)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4203f2260, 0x3b9aca00, 0xc42001a720)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Run
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109 +0x203
```

**What you expected to happen**:
calm

**How to reproduce it (as minimally and precisely as possible)**:
I do not know :(

**Anything else we need to know?**:
Only the Certificate and Issuer CustomResourceDefinitions were deployed, which caused one error, but after deploying the ClusterIssuer CRD and restarting the pod - panic:
```
I1026 19:56:16.066555       1 server.go:68] Listening on http://0.0.0.0:9402
I1026 19:56:16.068500       1 leaderelection.go:174] attempting to acquire leader lease...
I1026 19:56:16.091984       1 leaderelection.go:184] successfully acquired lease kube-system/cert-manager-controller
I1026 19:56:16.092199       1 factory.go:58] NS does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Issuer"}
I1026 19:56:16.092281       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 19:56:16.092342       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Certificate"}
I1026 19:56:16.092365       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Issuer"}
I1026 19:56:16.092379       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"ClusterIssuer"}
I1026 19:56:16.092392       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 19:56:16.092404       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"extensions", Version:"v1beta1", Kind:"Ingress"}
I1026 19:56:16.092437       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"ClusterIssuer"}
I1026 19:56:16.092457       1 factory.go:58] NS does not exist on lookup for kube-system - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 19:56:16.192761       1 logs.go:41] starting worker
I1026 19:56:16.192801       1 sync.go:81] Issuer my-app-tls-ci not ready
I1026 19:56:16.192859       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 19:56:16.192974       1 logs.go:41] starting worker
E1026 19:56:16.193111       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:290
/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:647
/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:321
/go/src/github.com/jetstack-experimental/cert-manager/pkg/issuer/acme/setup.go:59
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/sync.go:28
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:179
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:60
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:141
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:148
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2197
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6cb508]

goroutine 25 [running]:
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x126
panic(0x15ffe60, 0x2205470)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme.(*Client).doReg(0x0, 0x21ba740, 0xc4201b79c0, 0x0, 0x0, 0x17b3255, 0x3, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:647 +0x128
github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme.(*Client).GetReg(0x0, 0x21ba740, 0xc4201b79c0, 0x0, 0x0, 0x1, 0x14, 0xc42009c3c0)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:321 +0x7e
github.com/jetstack-experimental/cert-manager/pkg/issuer/acme.(*Acme).Setup(0xc420136400, 0x21ba740, 0xc4201b79c0, 0x21baa00, 0xc420136400, 0x0, 0x0, 0xc42014c0a0, 0xc42019d3e0)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/issuer/acme/setup.go:59 +0x390
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Sync(0xc420132320, 0x21ba740, 0xc4201b79c0, 0xc4202cea00, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/sync.go:28 +0xcc
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).processNextWorkItem(0xc420132320, 0x21ba740, 0xc4201b79c0, 0xc42019d3e0, 0x20, 0xc42067ed48, 0xc42022e4d0)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:179 +0x234
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).(github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.processNextWorkItem)-fm(0x21ba740, 0xc4201b79c0, 0xc42019d3e0, 0x20, 0x2, 0x2)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:60 +0x52
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).worker.func1(0xc420132320, 0xc4201c7860, 0x15749e0, 0xc4203baa60, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:141 +0x2ba
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).worker(0xc420132320, 0xc4201c7860)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:148 +0xe5
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Run.func1()
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109 +0x33
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4207ec020)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4207ec020, 0x3b9aca00, 0x0, 0x1, 0xc4201c7860)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4207ec020, 0x3b9aca00, 0xc4201c7860)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Run
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109 +0x203
```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
$ kubectl version --short
Client Version: v1.8.2
Server Version: v1.7.5+coreos.0
```
- Cloud provider or hardware configuration**:
GiantSwarm on AWS

`--v=6` as an arg, correct?  I can't tell from the logs.
```
I1026 20:11:05.998196       1 server.go:68] Listening on http://0.0.0.0:9402
I1026 20:11:05.998823       1 controller.go:103] Creating event broadcaster
I1026 20:11:05.999468       1 leaderelection.go:174] attempting to acquire leader lease...
I1026 20:11:06.015421       1 round_trippers.go:405] GET https://172.31.0.1:443/api/v1/namespaces/kube-system/endpoints/cert-manager-controller 200 OK in 15 milliseconds
I1026 20:11:06.026886       1 round_trippers.go:405] PUT https://172.31.0.1:443/api/v1/namespaces/kube-system/endpoints/cert-manager-controller 200 OK in 10 milliseconds
I1026 20:11:06.027060       1 leaderelection.go:184] successfully acquired lease kube-system/cert-manager-controller
I1026 20:11:06.027275       1 factory.go:58] NS does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Certificate"}
I1026 20:11:06.027303       1 controller.go:105] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"kube-system", Name:"cert-manager-controller", UID:"43d6e285-ba32-11e7-8276-021ebb038dfa", APIVersion:"v1", ResourceVersion:"3877263", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' my-app-tls-cert-manager-3963408647-kt075-external-cert-manager-controller became leader
I1026 20:11:06.027344       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Issuer"}
I1026 20:11:06.027365       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"ClusterIssuer"}
I1026 20:11:06.027383       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 20:11:06.027400       1 factory.go:66] existing informer does not exist on lookup for  - v1.GroupVersionKind{Group:"extensions", Version:"v1beta1", Kind:"Ingress"}
I1026 20:11:06.027438       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"ClusterIssuer"}
I1026 20:11:06.027456       1 factory.go:58] NS does not exist on lookup for kube-system - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 20:11:06.027495       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"certmanager.k8s.io", Version:"v1alpha1", Kind:"Issuer"}
I1026 20:11:06.027512       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 20:11:06.027537       1 controller.go:56] Starting shared informer factory
I1026 20:11:06.027574       1 controller.go:47] Starting clusterissuers controller
I1026 20:11:06.027588       1 controller.go:101] Starting clusterissuers control loop
I1026 20:11:06.027617       1 controller.go:47] Starting certificates controller
I1026 20:11:06.027632       1 controller.go:152] Starting certificates control loop
I1026 20:11:06.027730       1 controller.go:47] Starting issuers controller
I1026 20:11:06.027795       1 controller.go:99] Starting issuers control loop
I1026 20:11:06.028062       1 reflector.go:196] Starting reflector *v1.Secret (30s) from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028121       1 reflector.go:234] Listing and watching *v1.Secret from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028182       1 reflector.go:196] Starting reflector *v1alpha1.ClusterIssuer (30s) from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028225       1 reflector.go:234] Listing and watching *v1alpha1.ClusterIssuer from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028235       1 reflector.go:196] Starting reflector *v1beta1.Ingress (30s) from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028253       1 reflector.go:234] Listing and watching *v1beta1.Ingress from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028225       1 reflector.go:196] Starting reflector *v1alpha1.Certificate (30s) from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028349       1 reflector.go:234] Listing and watching *v1alpha1.Certificate from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028103       1 reflector.go:196] Starting reflector *v1.Secret (30s) from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028450       1 reflector.go:234] Listing and watching *v1.Secret from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028448       1 reflector.go:196] Starting reflector *v1alpha1.Issuer (30s) from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.028468       1 reflector.go:234] Listing and watching *v1alpha1.Issuer from github.com/jetstack-experimental/cert-manager/pkg/util/kube/factory.go:87
I1026 20:11:06.038935       1 round_trippers.go:405] GET https://172.31.0.1:443/apis/certmanager.k8s.io/v1alpha1/clusterissuers?resourceVersion=0 200 OK in 10 milliseconds
I1026 20:11:06.041283       1 round_trippers.go:405] GET https://172.31.0.1:443/api/v1/namespaces/kube-system/secrets?resourceVersion=0 200 OK in 13 milliseconds
I1026 20:11:06.044047       1 round_trippers.go:405] POST https://172.31.0.1:443/api/v1/namespaces/kube-system/events 201 Created in 16 milliseconds
I1026 20:11:06.044238       1 round_trippers.go:405] GET https://172.31.0.1:443/apis/certmanager.k8s.io/v1alpha1/certificates?resourceVersion=0 200 OK in 15 milliseconds
I1026 20:11:06.045681       1 round_trippers.go:405] GET https://172.31.0.1:443/apis/extensions/v1beta1/ingresses?resourceVersion=0 200 OK in 17 milliseconds
I1026 20:11:06.046270       1 round_trippers.go:405] GET https://172.31.0.1:443/apis/certmanager.k8s.io/v1alpha1/issuers?resourceVersion=0 200 OK in 17 milliseconds
I1026 20:11:06.063622       1 round_trippers.go:405] GET https://172.31.0.1:443/apis/certmanager.k8s.io/v1alpha1/certificates?resourceVersion=3864688&timeoutSeconds=349&watch=true 200 OK in 13 milliseconds
I1026 20:11:06.064450       1 round_trippers.go:405] GET https://172.31.0.1:443/api/v1/namespaces/kube-system/secrets?resourceVersion=3856071&timeoutSeconds=597&watch=true 200 OK in 16 milliseconds
I1026 20:11:06.065400       1 round_trippers.go:405] GET https://172.31.0.1:443/apis/certmanager.k8s.io/v1alpha1/clusterissuers?resourceVersion=3874850&timeoutSeconds=365&watch=true 200 OK in 26 milliseconds
I1026 20:11:06.068419       1 round_trippers.go:405] GET https://172.31.0.1:443/apis/certmanager.k8s.io/v1alpha1/issuers?resourceVersion=3864688&timeoutSeconds=331&watch=true 200 OK in 21 milliseconds
I1026 20:11:06.069175       1 round_trippers.go:405] GET https://172.31.0.1:443/apis/extensions/v1beta1/ingresses?resourceVersion=3855739&timeoutSeconds=333&watch=true 200 OK in 21 milliseconds
I1026 20:11:06.075968       1 round_trippers.go:405] GET https://172.31.0.1:443/api/v1/secrets?resourceVersion=0 200 OK in 47 milliseconds
I1026 20:11:06.107493       1 round_trippers.go:405] GET https://172.31.0.1:443/api/v1/secrets?resourceVersion=3856071&timeoutSeconds=577&watch=true 200 OK in 6 milliseconds
I1026 20:11:06.127656       1 shared_informer.go:116] caches populated
I1026 20:11:06.127701       1 logs.go:41] starting worker
I1026 20:11:06.127727       1 logs.go:41] starting worker
I1026 20:11:06.127840       1 shared_informer.go:116] caches populated
I1026 20:11:06.127854       1 controller.go:163] Synced all caches for certificates control loop
I1026 20:11:06.127871       1 controller.go:181] Starting certificates worker
I1026 20:11:06.127897       1 controller.go:199] certificates controller: syncing item 'default/my-app-tls-ci'
I1026 20:11:06.127915       1 sync.go:81] Issuer my-app-tls-ci not ready
I1026 20:11:06.127948       1 controller.go:201] certificates controller: error syncing item 'default/my-app-tls-ci': Issuer my-app-tls-ci not ready
I1026 20:11:06.127960       1 controller.go:210] Requeuing object due to error processing: Issuer my-app-tls-ci not ready
I1026 20:11:06.128002       1 shared_informer.go:116] caches populated
I1026 20:11:06.128039       1 controller.go:181] Starting certificates worker
I1026 20:11:06.128041       1 logs.go:41] starting worker
I1026 20:11:06.128061       1 logs.go:41] starting worker
I1026 20:11:06.128069       1 controller.go:140] issuers controller: syncing item 'default/my-app-tls-ci'
I1026 20:11:06.128076       1 controller.go:105] Event(v1.ObjectReference{Kind:"Certificate", Namespace:"default", Name:"my-app-tls-ci", UID:"29d0278d-ba75-11e7-8276-021ebb038dfa", APIVersion:"certmanager.k8s.io/v1alpha1", ResourceVersion:"3855672", FieldPath:""}): type: 'Warning' reason: 'ErrorIssuerNotReady' Issuer my-app-tls-ci not ready
I1026 20:11:06.128115       1 factory.go:70] existing informer found for  - v1.GroupVersionKind{Group:"", Version:"v1", Kind:"Secret"}
I1026 20:11:06.128133       1 setup.go:36] my-app-tls-ci: getting acme account private key 'default/{{my-app-tls-ci-acme} }'
I1026 20:11:06.128149       1 acme.go:85] getting private key (my-app-tls-ci-acme->tls.key) for acme issuer default/my-app-tls-ci
I1026 20:11:06.128160       1 setup.go:54] my-app-tls-ci: Failed to verify ACME account: error decoding private key PEM block
I1026 20:11:06.128179       1 setup.go:58] my-app-tls-ci: verifying existing registration with ACME server
E1026 20:11:06.128288       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:290
/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:647
/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:321
/go/src/github.com/jetstack-experimental/cert-manager/pkg/issuer/acme/setup.go:59
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/sync.go:28
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:179
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:60
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:141
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:148
/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2197
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6cb508]

goroutine 194 [running]:
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x126
panic(0x15ffe60, 0x2205470)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme.(*Client).doReg(0x0, 0x21ba740, 0xc42009a380, 0x0, 0x0, 0x17b3255, 0x3, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:647 +0x128
github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme.(*Client).GetReg(0x0, 0x21ba740, 0xc42009a380, 0x0, 0x0, 0x1, 0x14, 0xc4203ab310)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/golang.org/x/crypto/acme/acme.go:321 +0x7e
github.com/jetstack-experimental/cert-manager/pkg/issuer/acme.(*Acme).Setup(0xc420404a80, 0x21ba740, 0xc42009a380, 0x21baa00, 0xc420404a80, 0x0, 0x0, 0xc420177b80, 0xc4207d8240)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/issuer/acme/setup.go:59 +0x390
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Sync(0xc420353400, 0x21ba740, 0xc42009a380, 0xc4207c2000, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/sync.go:28 +0xcc
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).processNextWorkItem(0xc420353400, 0x21ba740, 0xc42009a380, 0xc4207d8240, 0x20, 0x2, 0x2)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:179 +0x234
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).(github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.processNextWorkItem)-fm(0x21ba740, 0xc42009a380, 0xc4207d8240, 0x20, 0x2, 0x2)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:60 +0x52
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).worker.func1(0xc420353400, 0xc4200f5e00, 0x15749e0, 0xc42020d490, 0x0, 0x0)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:141 +0x2ba
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).worker(0xc420353400, 0xc4200f5e00)
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:148 +0xe5
github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Run.func1()
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109 +0x33
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc42028b220)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc42028b220, 0x3b9aca00, 0x0, 0x1, 0xc4200f5e00)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc42028b220, 0x3b9aca00, 0xc4200f5e00)
	/go/src/github.com/jetstack-experimental/cert-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/jetstack-experimental/cert-manager/pkg/controller/issuers.(*Controller).Run
	/go/src/github.com/jetstack-experimental/cert-manager/pkg/controller/issuers/controller.go:109 +0x203
```

--- a/pkg/issuer/acme/setup.go
+++ b/pkg/issuer/acme/setup.go
@@ -13,6 +13,7 @@ import (
 
 	"github.com/golang/glog"
 	"github.com/jetstack-experimental/cert-manager/pkg/apis/certmanager/v1alpha1"
+	"github.com/jetstack-experimental/cert-manager/pkg/util/errors"
 	"github.com/jetstack-experimental/cert-manager/pkg/util/kube"
 	"github.com/jetstack-experimental/cert-manager/pkg/util/pki"
 )
@@ -33,7 +34,7 @@ const (
 func (a *Acme) Setup(ctx context.Context) error {
 	glog.V(4).Infof("%s: getting acme account private key '%s/%s'", a.issuer.GetObjectMeta().Name, a.resourceNamespace, a.issuer.GetSpec().ACME.PrivateKey.Name)
 	cl, err := a.acmeClient()
-	if k8sErrors.IsNotFound(err) {
+	if k8sErrors.IsNotFound(err) || errors.IsInvalidData(err) {
 		glog.V(4).Infof("%s: generating acme account private key '%s/%s'", a.issuer.GetObjectMeta().Name, a.resourceNamespace, a.issuer.GetSpec().ACME.PrivateKey.Name)
 		var accountPrivKey *rsa.PrivateKey
 		accountPrivKey, err = a.createAccountPrivateKey()
@@ -51,6 +52,7 @@ func (a *Acme) Setup(ctx context.Context) error {
 		s := messageAccountVerificationFailed + err.Error()
 		glog.V(4).Infof("%s: %s", a.issuer.GetObjectMeta().Name, s)
 		a.recorder.Event(a.issuer, v1.EventTypeWarning, errorAccountVerificationFailed, s)
+		return err
 	}
 
 	glog.V(4).Infof("Verifying ")

